mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,449 @@
+            for artifact in client.list_artifacts(run_id=run.info.run_id):
+                # Get the artifact's local path (MLFlow suggests that if the artifact is already in the local filesystem
+                # its local path will be returned:
+                artifact_local_path = mlflow.artifacts.download_artifacts(
+                    run_id=run.info.run_id,
+                    artifact_path=artifact.path,
+                )
+                # Check if the artifact is a model (will be logged after the artifacts):
+                if artifact.is_dir and os.path.exists(
+                    os.path.join(artifact_local_path, "MLmodel")
+                ):
+                    model_paths.append(artifact_local_path)
+                else:
+                    # Log the artifact:
+                    artifact = MLFlowTracker._log_artifact(
+                        context_or_project=context,
+                        key=pathlib.Path(artifact.path).name.replace(".", "_"),
+                        local_path=artifact_local_path,
+                        tmp_path=tmp_dir,
+                    )
+                    artifacts[artifact.key] = artifact
+            if model_paths:
+                for model_path in model_paths:
+                    MLFlowTracker._log_model(
+                        context_or_project=context,
+                        model_uri=model_path,
+                        key=pathlib.Path(model_path).stem,
+                        metrics=results,
+                        extra_data=artifacts,
+                        tmp_path=tmp_dir,
+                    )
+
+    @staticmethod
+    def _log_model(
+        context_or_project: Union[MLClientCtx, MlrunProject],
+        model_uri: str,
+        key: str,
+        metrics: dict,
+        extra_data: dict,
+        tmp_path: os.path,
+    ):
+        """
+        Log the given produced model from MLFlow as a model artifact in MLRun.
+
+        :param context_or_project: The MLRun context or project to log to.
+        :param model_uri:          The local path to the model (an MLFlow model directory locally downloaded).
+        :param key:                The model artifact's key.
+        :param metrics:            The key/value dict of model metrics
+        :param extra_data:         The extra data to log in addition to the model (training data for example)
+        :param tmp_path:           The path to the dir where we temporarily save model and artifacts
+        """
+        # Get the model info from MLFlow:
+        model_info = mlflow.models.get_model_info(model_uri=model_uri)
+
+        # Prepare the archive path:
+        model_uri = pathlib.Path(model_uri)
+        archive_path = pathlib.Path(tmp_path) / f"{model_uri.stem}.zip"
+
+        # Zip the artifact:
+        with zipfile.ZipFile(archive_path, "w") as zip_file:
+            for path in model_uri.rglob("*"):
+                zip_file.write(filename=path, arcname=path.relative_to(model_uri))
```

Comment:
How large will these models be? We're zipping them which may be extremely expensive if the model is very large. Also uploading the model to the artifact store will be time-consuming etc.
I'm thinking we might want to impose some limits on the actual size of models that we support, maybe through some config param.

Reply:
they can be very big i guess, 
if currently there is no limitation on uploadind to the artifact store how do you deal with this?

Reply:
the models size is determined by the user and if it's large and takes a lot of time and resources it's on him

Reply:
If we don't want to add limits we need to have an indication of what's going on.
How about a progress bar, maybe we can use tqdm here?